### PR TITLE
faster rendering by looking at min and max scales before read

### PR DESCRIPTION
**Pull request by sloob**
To increase the performance of the GridCoverageReaderRenderer, raster rendering will only be performed if the map's current scale is within the min / max scale of the layer.

Signed-off-by: sloob <sebastian.loob@ibykus.de>

### CODE REVIEW

**Comment by fgdrf** [comment]
started ci-build : https://ci.eclipse.org/udig/job/uDig-PR/30/

File: plugins/org.locationtech.udig.render.gridcoverage.basic/src/org/locationtech/udig/render/internal/gridcoverage/basic/GridCoverageReaderRenderer.java [hunk 1]
```diff
@@ -111,6 +111,11 @@ public GridCoverageReaderRenderer(){
     public synchronized void render( Graphics2D graphics, IProgressMonitor monitor )
             throws RenderException {
         try {
+                // faster rendering if out of scale
+                GridCoverageRenderState state = getRenderState(getContext());
+                double scale = state.context.getViewportModel().getScaleDenominator();
+                if (scale < state.minScale || scale > state.maxScale)
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.render.gridcoverage.basic/src/org/locationtech/udig/render/internal/gridcoverage/basic/GridCoverageReaderRenderer.java]: Great, change looks good! Any chance to add a test case?

**Comment by fgdrf** [comment]
Started a CI-Build for this pull request again : https://ci.eclipse.org/udig/job/uDig-PR/32/

**Comment by fgdrf** [comment]
Started a CI-Build for this pull request again : https://ci.eclipse.org/udig/job/uDig-PR/32/

**Comment by fgdrf** [comment]
Started a CI-Build for this pull request again : https://ci.eclipse.org/udig/job/uDig-PR/32/

File: plugins/org.locationtech.udig.render.gridcoverage.basic/src/org/locationtech/udig/render/internal/gridcoverage/basic/GridCoverageReaderRenderer.java (hunk 1 above)
Comment (fgdrf) [reply plugins/org.locationtech.udig.render.gridcoverage.basic/src/org/locationtech/udig/render/internal/gridcoverage/basic/GridCoverageReaderRenderer.java]: IMH minScale / maxScale of state object might be null, if one or the other is not set.

Can you explaint how this change is related to line 232/ 227 where Rule of Style is compared to teh currentScale? I guess it is possible to define different rules for the same resource and one rule might fit where another doesn't.

So maybe it makes sense to move the style rule scale comparence before creating a coverage ..

File: plugins/pom.xml [hunk 2]
```diff
@@ -33,6 +33,7 @@
                 <module>org.locationtech.udig.libs.tests</module>
                 <module>org.locationtech.udig.location.test</module>
                 <module>org.locationtech.udig.render.feature.basic.test</module>
+				<module>org.locationtech.udig.render.gridcoverage.basic.test</module>
```
Comment (fgdrf) [inline plugins/pom.xml]: Looks like tabs are at the beginning of the lin rather that spaces, can we fix it?

File: plugins/org.locationtech.udig.render.gridcoverage.basic.test/src/org/locationtech/udig/render/internal/gridcoverage/basic/BasicGridCoverage2DTestReader.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                return null;
+            }
+
+            @Override
+            public ParameterValueGroup getReadParameters() {
+                Map<String, Object> paramMap = new HashMap<String, Object>();
+                paramMap.put("name", "BasicGridCoverage2DFormat");  //$NON-NLS-1$//$NON-NLS-2$
+                paramMap.put(AbstractGridFormat.READ_GRIDGEOMETRY2D.getName().toString(),new GridGeometry2D(new Rectangle(), new Rectangle()));
+                
+/*                GridEnvelope2D gridEnvelope = new GridEnvelope2D(0, 0, mapDisplay.getWidth(), mapDisplay.getHeight());
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.render.gridcoverage.basic.test/src/org/locationtech/udig/render/internal/gridcoverage/basic/BasicGridCoverage2DTestReader.java]: Do we need the commented lines on intial commit? Are you kind to remove these?

Comment (sloob) [reply plugins/org.locationtech.udig.render.gridcoverage.basic.test/src/org/locationtech/udig/render/internal/gridcoverage/basic/BasicGridCoverage2DTestReader.java]: Sorry for the bad code formatting. Hope everything is fine now :)

**Comment by fgdrf** [comment]
Looks good! Build-Job failure hasn't to do anything with your changes (https://ci.eclipse.org/udig/job/uDig-PR/38/)

Many thanks again for this improvement

**Pull request merged**